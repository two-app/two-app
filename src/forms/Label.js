import React from "react";
import {Text, StyleSheet} from "react-native";
import PropTypes from "prop-types";
import Colors from "../Colors";

const Label = ({text}) => <Text style={styles.label}>{text}</Text>;

Label.propTypes = {
    text: PropTypes.string.isRequired
};

const styles = StyleSheet.create({
    label: {
        marginTop: 20,
        fontSize: 12,
        color: Colors.DARK,
        fontWeight: "bold"
    }
});

export default Label;