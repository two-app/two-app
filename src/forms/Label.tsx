import {StyleSheet, Text} from 'react-native';

import Colors from '../Colors';

export const Label = ({text}: {text: string}) => (
  <Text style={styles.label}>{text}</Text>
);

const styles = StyleSheet.create({
  label: {
    marginTop: 20,
    fontSize: 12,
    color: Colors.DARK,
    fontWeight: 'bold',
  },
});
