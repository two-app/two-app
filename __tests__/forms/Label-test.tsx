import "react-native";
import React from "react";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import type { ShallowWrapper } from "enzyme";
import { shallow } from "enzyme";

import Label from "../../src/forms/Label";

test("should maintain snapshot", () =>
  expect(renderer.create(<Label text={"Test Label"} />)).toMatchSnapshot());

describe("Render Tests", () => {
  const label = "Test Label Text";
  let wrapper: ShallowWrapper;

  beforeEach(() => (wrapper = shallow(<Label text={label} />)));

  test("it should create a text element", () =>
    expect(wrapper.exists("Text")).toBe(true));

  test("it should render the label text", () =>
    expect(wrapper.find("Text").render().text()).toContain(label));

  test("it should style the label", () => {
    const appliedStyle = wrapper.find("Text").prop("style");
    expect(appliedStyle).toHaveProperty("fontWeight", "bold");
  });
});
