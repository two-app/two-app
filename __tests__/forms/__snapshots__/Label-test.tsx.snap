// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should maintain snapshot 1`] = `
<Text
  style={
    Object {
      "color": "rgb(37, 37, 37)",
      "fontSize": 12,
      "fontWeight": "bold",
      "marginTop": 20,
    }
  }
>
  Test Label
</Text>
`;
